inference_perf/client/storage/gcs.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from google.cloud import storage
+from google.cloud.exceptions import GoogleCloudError
+from pydantic import BaseModel
+from inference_perf.client.storage.base import StorageClient
+
+
+class GoogleCloudStorageConfig(BaseModel):
+    bucket_name: str
+
+
+class GoogleCloudStorageClient(StorageClient):
+    def __init__(self, config: GoogleCloudStorageConfig) -> None:
+        super().__init__()
+        print("Created new GCS client")
+        self.output_bucket = config.bucket_name
+        self.client = storage.Client()
+
+        self.bucket = self.client.lookup_bucket(config.bucket_name)
+        if self.bucket is None:
+            raise ValueError(f"GCS bucket '{config.bucket_name}' does not exist or is inaccessible.")
+
+    def push_report(self, reports: List[str]) -> None:
+        for index, content in enumerate(reports):
+            filename = f"report-{index}"
+            blob_path = f"{self.path}/{filename}" if self.path else filename
```

Comment:
We should make self.path optionally configurable instead of only base class `reports-timestamp`. Or at least the ability to specify a prefix so reports are written as `bucket/prefix/reports-timestamp/reports-index`